### PR TITLE
Add test for get tokens function

### PR DESCRIPTION
**Pull request by stringtheory1234**

### CODE REVIEW

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 1]
```diff
@@ -70,4 +77,20 @@ public void testGetWaitingTime() throws Exception {
         
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:81]: Follow the AAA pattern
https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 2]
```diff
@@ -70,4 +77,20 @@ public void testGetWaitingTime() throws Exception {
...
+    public void testGetTokens() throws Exception {
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:82]: rename testGetTokens_success

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 3]
```diff
@@ -70,4 +77,20 @@ public void testGetWaitingTime() throws Exception {
...
+        UUID shopId = UUID.randomUUID();
+        Shop shop = new Shop();
+        shop.setStatus(ShopStatus.ACTIVE);
+        inMemoryVirtualQueueRepository.getShopMap().put(shopId, shop);
+        
+        List<Token> tokens = new ArrayList<>();
+        tokens.add(new Token(UUID.randomUUID(), shopId, 1, Status.ACTIVE));
+        inMemoryVirtualQueueRepository.getShopIdToListOfTokensMap().put(shopId, tokens);
+
+        List<Token> getTokensResponse = inMemoryVirtualQueueRepository.getTokens(shopId);
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:92]: rename expectedGetTokensResponseList

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 4]
```diff
@@ -70,4 +77,19 @@ public void testGetWaitingTime() throws Exception {
         
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetTokens_success() throws Exception {
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:82]: Get tokens for a shop right? Make that evident in the name

Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 3 above)
Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 5]
```diff
@@ -70,4 +94,19 @@ public void testGetWaitingTime() throws Exception {
         
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetTokens_forShop_success() throws Exception {
+        // Arrange
+        // Add Shop
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:101]: Not adding any value. Can be removed? Add comment only if you feel it'll add some additional value.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 6]
```diff
@@ -70,4 +94,19 @@ public void testGetWaitingTime() throws Exception {
...
+        UUID shopId = AddShopToRepository();
+        // Add list of tokens
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:103]: same

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 7]
```diff
@@ -50,12 +57,29 @@ public void setUp(){
         inMemoryVirtualQueueRepository.getShopOwnerMap().clear();
     }
 
+    public UUID AddShopToRepository(){
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:60]: addShopToRepository

Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:60]: can be private?

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 8]
```diff
@@ -50,12 +57,29 @@ public void setUp(){
...
+        Shop shop = new Shop(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+        shop.setStatus(ShopStatus.ACTIVE);
+        UUID shopId = UUID.randomUUID();
+        inMemoryVirtualQueueRepository.getShopMap().put(shopId, shop);
+        return shopId;
+    }
+
+    public void AddTokenListToShop(UUID shopId){
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:68]: private?

Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:68]: addTokenListToShop?

Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 7 above)
Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 5 above)
Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: yes removed!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 7 above)
Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 6 above)
Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 9]
```diff
@@ -50,12 +57,29 @@ public void setUp(){
         inMemoryVirtualQueueRepository.getShopOwnerMap().clear();
     }
 
+    private UUID addShopToRepository(){
+        Shop shop = new Shop(SHOP_OWNER_ID, SHOP_NAME, SHOP_ADDRESS, PHONE_NUMBER, SHOP_TYPE);
+        shop.setStatus(ShopStatus.ACTIVE);
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:62]: Can these 2 lines be merged?

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 10]
```diff
@@ -50,12 +57,29 @@ public void setUp(){
         inMemoryVirtualQueueRepository.getShopOwnerMap().clear();
     }
 
+    private UUID addShopToRepository(){
```
Comment (priytanvani) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:60]: Nit. Add ' ' before '{'.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 9 above)
Comment (priytanvani) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:62]: With our current design, don't think so.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java (hunk 10 above)
Comment (priytanvani) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:60]: Nit. All helper methods should go in the end after all the tests.

Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

Comment (stringtheory1234) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java]: done!

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 11]
```diff
@@ -52,22 +59,56 @@ public void setUp(){
 
     @Test
     public void testCreateShop() throws Exception {
+        // Arrange
```
Comment (priytanvani) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:62]: Nit. Remember to add '.' at the end of every comment.

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 12]
```diff
@@ -52,22 +59,56 @@ public void setUp(){
 
     @Test
     public void testCreateShop() throws Exception {
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:61]: _success

File: backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java [hunk 13]
```diff
@@ -52,22 +59,56 @@ public void setUp(){
...
     @Test
     public void testGetWaitingTime() throws Exception {
         long waitingTime = inMemoryVirtualQueueRepository.getWaitingTime();
-        
+
         assertEquals("Waiting Time per customer is", WAITING_TIME_MINS, waitingTime);
     }
+
+    @Test
+    public void testGetTokens_forShop_success() throws Exception {
```
Comment (sdumka) [inline backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:88]: one word

Comment (sdumka) [reply backendservice/src/test/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepositoryTest.java:88]: testGetTokensForShop_success

**Pull request merged**